discussions/api.py

```diff
@@ -90,7 +90,10 @@ def create_discussion_user(discussion_user):
     Raises:
         DiscussionUserSyncException: if there was an error syncing the profile
     """
-    profile = discussion_user.user.profile
+    try:
```

Comment:
It seems strange to me to have the code raise a `DiscussionUserSyncException` if the API call fails but silence this error. It seems like we would want to raise an error since it's an error case, we just don't want it to halt the processing in the task or fill sentry with useless errors. What do you think of filtering out the missing profile and missing discussion username cases in `get_membership_ids` instead?

Reply:
@noisecapella I added code to filter out in `get_membership_ids`, the missing username should no longer be a case now too.